pkg/types/ruletypes/threshold.go

```diff
@@ -110,6 +131,39 @@ func (r BasicRuleThresholds) ShouldAlert(series v3.Series, unit string) (Vector,
 	}
 	return resultVector, nil
 }
+func (r BasicRuleThresholds) Eval(series v3.Series, unit string, evalData EvalData) (Vector, error) {
+	var resultVector Vector
+	thresholds := []BasicRuleThreshold(r)
+	sortThresholds(thresholds)
+	for _, threshold := range thresholds {
+		smpl, shouldAlert := threshold.shouldAlert(series, unit)
+		if shouldAlert {
+			smpl.Target = *threshold.TargetValue
+			smpl.TargetUnit = threshold.TargetUnit
+			resultVector = append(resultVector, smpl)
+			continue
+		}
+
+		// Prepare alert hash from series labels and threshold name if recovery target option was provided
+		if threshold.RecoveryTarget == nil {
+			continue
+		}
+		sampleLabels := PrepareSampleLabelsForRule(series.Labels, threshold.Name)
+		alertHash := sampleLabels.Hash()
+		// check if alert is active and then check if recovery threshold matches
+		if evalData.HasActiveAlert(alertHash) {
+			smpl, matchesRecoveryThrehold := threshold.matchesRecoveryThreshold(series, unit)
+			if matchesRecoveryThrehold {
+				smpl.Target = *threshold.RecoveryTarget
+				smpl.TargetUnit = threshold.TargetUnit
```

Comment:
It doesn't make sense to set the `Target` with the `RecoveryTarget` value. Let's introduce the separate var for it.